packages/@aws-cdk/aws-ecs-patterns/lib/base/queue-processing-service-base.ts

```diff
@@ -273,8 +287,20 @@ export abstract class QueueProcessingServiceBase extends CoreConstruct {
     this.secrets = props.secrets;
 
     // Determine the desired task count (minimum) and maximum scaling capacity
-    this.desiredCount = props.desiredTaskCount ?? 1;
-    this.maxCapacity = props.maxScalingCapacity || (2 * this.desiredCount);
+    if (!this.node.tryGetContext(cxapi.ECS_REMOVE_DEFAULT_DESIRED_COUNT)) {
+      this.desiredCount = props.desiredTaskCount != null ? props.desiredTaskCount : 1;
+      this.minCapacity = props.minScalingCapacity || this.desiredCount;
+      this.maxCapacity = props.maxScalingCapacity || (2 * this.desiredCount);
+    } else {
+      this.desiredCount = props.desiredTaskCount;
+      if (this.desiredCount != null) {
+        this.minCapacity = props.minScalingCapacity || this.desiredCount;
+        this.maxCapacity = props.maxScalingCapacity || (2 * this.desiredCount);
+      } else {
+        this.minCapacity = props.minScalingCapacity || 1;
+        this.maxCapacity = props.maxScalingCapacity || 2;
+      }
+    }
```

Comment:
Are there unit tests for this logic?

Reply:
Yes and new integ tests as well

Reply:
I'm not seeing any mention of `minCapacity`or `maxCapacity` being passed via props in any of the unit test or new integ test files -- could you point me to where?

Reply:
Added more tests to test each use case.